src/wp-includes/formatting.php

```diff
@@ -3914,10 +3914,22 @@ function wp_trim_excerpt( $text = '', $post = null ) {
 		$text = strip_shortcodes( $text );
 		$text = excerpt_remove_blocks( $text );
 
+		/*
+		 * Temporarily unhook wp_filter_content_tags() since any tags
+		 * within the excerpt are stripped out. Modifying the tags here
+		 * is wasteful and can lead to bugs in the image counting logic.
+		 */
+		$filter_removed = remove_filter( 'the_content', 'wp_filter_content_tags' );
+
 		/** This filter is documented in wp-includes/post-template.php */
 		$text = apply_filters( 'the_content', $text );
 		$text = str_replace( ']]>', ']]&gt;', $text );
 
+		// Only restore the filter callback if it was removed above.
+		if ( $filter_removed ) {
+			add_filter( 'the_content', 'wp_filter_content_tags' );
```

Comment:
Someone could have changed the priority of this filter intentionally, so we should put it back with the priority exactly how we found it.

```suggestion
			add_filter( 'the_content', 'wp_filter_content_tags', $filter_removed );
```

Reply:
`$filter_removed` is a boolean. If we want to `$priority`, we will have to do something like this. 

https://github.com/WordPress/wordpress-develop/pull/4439/files#r1194977298

Reply:
This was discussed here: https://github.com/WordPress/wordpress-develop/pull/4439#discussion_r1194653054

Note that `remove_filter()` doesn't return the priority, instead you have to know the priority when you call that function.

Reply:
Right, sorry for missing the return value difference in my previous comment. From a strict backwards compatibility perspective, I think we need to reset the filter to the same priority in which we found it since plugins could be intentionally moving the priority of this function for any reason.

Reply:
I'm not sure this will be necessary since the the third argument to `remove_filter()` is `$priority` and has a default value of `10`. So this means that if `$filter_removed` is `true`, the filter had a priority of `10` and which is the default in `add_filter()` here too.

Reply:
@joemcgill I'd prefer to leave this as is, as plugins shouldn't change the priorities of `the_content` filter - it'll likely break other plugins. More importantly, the way this is currently here is not a backward compatibility break since, if a plugin has changed it, we wouldn't unhook and rehook the filter in any case - effectively, if a plugin has changed that filter, they wouldn't get the benefits of this fix, but nothing would actually break / get worse than it was before.

But I don't feel too strongly about that, so happy to adjust.

Reply:
Ah, that's a good point that we're removing it from a specific priority via the default param. If someone moved the priority themselves then this workaround wouldn't help them at all. Happy to leave it as is then, though adding a note to the inline comment above where we're restoring the filter might be nice.

Reply:
Clarified in a comment in https://github.com/WordPress/wordpress-develop/pull/4439/commits/49a8a52cda65a60d5d2e2df973c3390a4dbd1490.